File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/model/erd/ApplyBean.java
package org.javamaster.invocationlab.admin.model.erd;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author yudong
 * @date 2023/2/12
 */
@NoArgsConstructor
@Data
public class ApplyBean {
    @JsonProperty("JAVA")
    private JAVABean jAVA;
    @JsonProperty("MYSQL")
    private MYSQLBean mYSQL;
    @JsonProperty("ORACLE")
    private ORACLEBean oRACLE;
    @JsonProperty("SQLServer")
    private SQLServerBean sQLServer;
    @JsonProperty("PostgreSQL")
    private PostgreSQLBean postgreSQL;

    @NoArgsConstructor
    @Data
    public static class JAVABean {
        private String type;
    }

    @NoArgsConstructor
    @Data
    public static class MYSQLBean {
        private String type;
    }

    @NoArgsConstructor
    @Data
    public static class ORACLEBean {
        private String type;
    }

    @NoArgsConstructor
    @Data
    public static class SQLServerBean {
        private String type;
    }

    @NoArgsConstructor
    @Data
    public static class PostgreSQLBean {
        private String type;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/model/erd/ApplyBean.kt
package org.javamaster.invocationlab.admin.model.erd

import org.javamaster.invocationlab.admin.annos.AllOpen

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @author yudong
 * @date 2023/2/12
 */


@AllOpen
class ApplyBean {
    @JsonProperty("JAVA")
    var jAVA: JAVABean? = null

    @JsonProperty("MYSQL")
    var mYSQL: MYSQLBean? = null

    @JsonProperty("ORACLE")
    var oRACLE: ORACLEBean? = null

    @JsonProperty("SQLServer")
    var sQLServer: SQLServerBean? = null

    @JsonProperty("PostgreSQL")
    var postgreSQL: PostgreSQLBean? = null


    @AllOpen
    class JAVABean {
        var type: String? = null
    }


    @AllOpen
    class MYSQLBean {
        var type: String? = null
    }


    @AllOpen
    class ORACLEBean {
        var type: String? = null
    }


    @AllOpen
    class SQLServerBean {
        var type: String? = null
    }


    @AllOpen
    class PostgreSQLBean {
        var type: String? = null
    }
}
